pkgs/development/python-modules/ansible-compat/default.nix

```diff
@@ -27,6 +28,15 @@ buildPythonPackage rec {
     hash = "sha256-n/ICReG9nemyOjZ5AlJKsOEfvPt0GDGZbaXaW2Crld8=";
   };
 
+  patches = [
+    # Patch incompatibility between tests and ansible-core
+    # TODO: remove when new release is out
+    (fetchpatch2 {
+      url = "https://github.com/ansible/ansible-compat/commit/39729217f41282884414c4b19f3c5156383fedc5.patch";
```

Comment:
I think with fetchatch2 you need to add `?full_index=1` to the end of the url. https://github.com/NixOS/nixpkgs/issues/266556#issuecomment-1837651068